rules/rulesets.py

```diff
@@ -1,49 +1,90 @@
-from .predicates import predicate
+from .predicates import predicate, Predicate
+from inspect import isfunction
 
 
 class RuleSet(dict):
+    def __init__(self, *args, **kwargs):
+        self.update(*args, **kwargs)
+
     def test_rule(self, name, *args, **kwargs):
+        # return name in self and self[name]['pred'].test(*args, **kwargs)
         return name in self and self[name].test(*args, **kwargs)
 
     def rule_exists(self, name):
         return name in self
 
-    def add_rule(self, name, pred):
+    def rule_verbose_name(self, name):
+        return self["get_%s_display" % name] or name
```

Comment:
```suggestion
        return self["name"]["verbose_name"]
```

This must return None if no verbose name has been provided for the rule and callers can choose whether to fallback to `name`.